bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/handler/PulseaudioBridgeHandler.java

```diff
@@ -189,4 +218,41 @@ public boolean unregisterDeviceStatusListener(DeviceStatusListener deviceStatusL
     public void bindingConfigurationChanged() {
         update();
     }
+
+    public void resetKnownActiveDevices() {
+        lastActiveDevices = new HashSet<>();
+        update();
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof PulseaudioHandler) {
+            this.childHandlersInitialized.add((PulseaudioHandler) childHandler);
+        } else {
+            logger.error("This bridge can only support PulseaudioHandler child");
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        this.childHandlersInitialized.remove(childHandler);
+    }
+
+    public void waitForInitialization() {
+        if (initializationCompleted) {
+            logger.debug("No need to wait for initialization");
+            return;
+        }
+        try {
+            logger.debug("Waiting for initialization end");
+            initializationCompletedLock.lock();
+            if (!initializationCompleted && !initializationCompletedCondition.await(10, TimeUnit.SECONDS)) {
```

Comment:
If I am not wrong, in case one thing handler is calling waitForInitialization before the bridge handler finished the execution of update(), it will lock the end of update() until the 10 seconds expires, and then you are still not sure that the update() will be unlocked as the lock could be taken by another thing..

Reply:
It worked but it was a bad usage.
See below for an update.